nimble/host/include/host/ble_audio_common.h

```diff
@@ -0,0 +1,179 @@
+
+#define BLE_AUDIO_LOCATION_FRONT_LEFT                             BIT(0)
+#define BLE_AUDIO_LOCATION_FRONT_RIGHT                            BIT(1)
+#define BLE_AUDIO_LOCATION_FRONT_CENTER                           BIT(2)
+#define BLE_AUDIO_LOCATION_LOW_FREQ_EFFECTS_1                     BIT(3)
+#define BLE_AUDIO_LOCATION_BACK_LEFT                              BIT(4)
+#define BLE_AUDIO_LOCATION_FRONT_LEFT_CENTER                      BIT(5)
+#define BLE_AUDIO_LOCATION_FRONT_RIGHT_CENTER                     BIT(6)
+#define BLE_AUDIO_LOCATION_BACK_CENTER                            BIT(7)
+#define BLE_AUDIO_LOCATION_LOW_FREQ_EFFECTS_2                     BIT(8)
+#define BLE_AUDIO_LOCATION_SIDE_LEFT                              BIT(9)
+#define BLE_AUDIO_LOCATION_SIDE_RIGHT                             BIT(10)
+#define BLE_AUDIO_LOCATION_TOP_FRONT_LEFT                         BIT(11)
+#define BLE_AUDIO_LOCATION_TOP_FRONT_RIGHT                        BIT(12)
+#define BLE_AUDIO_LOCATION_TOP_FRONT_CENTER                       BIT(13)
+#define BLE_AUDIO_LOCATION_TOP_CENTER                             BIT(14)
+#define BLE_AUDIO_LOCATION_TOP_BACK_LEFT                          BIT(15)
+#define BLE_AUDIO_LOCATION_TOP_BACK_RIGHT                         BIT(16)
+#define BLE_AUDIO_LOCATION_TOP_SIDE_LEFT                          BIT(17)
+#define BLE_AUDIO_LOCATION_TOP_SIDE_RIGHT                         BIT(18)
+#define BLE_AUDIO_LOCATION_TOP_BACK_CENTER                        BIT(19)
+#define BLE_AUDIO_LOCATION_BOTTOM_FRONT_CENTER                    BIT(20)
+#define BLE_AUDIO_LOCATION_BOTTOM_FRONT_LEFT                      BIT(21)
+#define BLE_AUDIO_LOCATION_BOTTOM_FRONT_RIGHT                     BIT(22)
+#define BLE_AUDIO_LOCATION_LEFT_SURROUND                          BIT(23)
+#define BLE_AUDIO_LOCATION_RIGHT_SURROUND                         BIT(24)
+
+#define BLE_AUDIO_CODEC_SAMPLING_FREQ_TYPE                        0x01
+#define BLE_AUDIO_CODEC_FRAME_DURATION_TYPE                       0x02
+#define BLE_AUDIO_CODEC_AUDIO_CHANNEL_ALLOCATION_TYPE             0x03
+#define BLE_AUDIO_CODEC_OCTETS_PER_CODEC_FRAME_TYPE               0x04
+#define BLE_AUDIO_CODEC_FRAME_BLOCKS_PER_SDU_TYPE                 0x05
+
+/**
+ * @brief Helper macro used to build LTV array of Codec_Specific_Configuration.
+ *
+ * @param _sampling_freq               Sampling_Frequency - single octet value
+ * @param _frame_duration              Frame_Duration - single octet value
+ * @param _audio_channel_alloc         Audio_Channel_Allocation -
+ *                                     four octet value
+ * @param _octets_per_codec_frame      Octets_Per_Codec_Frame -
+ *                                     two octet value
+ * @param _codec_frame_blocks_per_sdu  Codec_Frame_Blocks_Per_SDU -
+ *                                     single octet value
+ *
+ * @return          Pointer to a `ble_uuid16_t` structure.
+ */
+#define BLE_AUDIO_BUILD_CODEC_CONFIG(_sampling_freq,                          \
```

Comment:
I noticed that this does not behave as expected: I'll rework this macro to fit spec better